keps/sig-network/4969-cluster-domain-pod-api/README.md

```diff
@@ -0,0 +1,690 @@
+   lookup times (and DNS server load), because it requires clients to try each
+   option separately.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+#### Split Domains
+
+Kubernetes currently does not prohibit different kubelets from specifying 
+different cluster domains (`node-a` could set `cluster.local` while `node-b` 
+specifies `cluster.remote`).
+Advertising FQDNs generated using this API could cause issues in these mixed
+environments, since `node-b` might not be able to resolve `cluster.local`
+FQDNs correctly.
+
+## Design Details
+
+A new field `dns.clusterDomain` would be added to `PodStatus`:
+
+```go
+type PodStatus struct {
+    // ..existing fields elided
+    DNS *PodDNSStatus
+}
+
+type PodDNSStatus struct {
+    ClusterDomain *string
+}
+```
+
+The field would be populated by the Kubelet during pod initialization.
+
+The field would also be accessible from the Downward API via `fieldRef`.
+
+@thockin originally suggested that `clusterDomain` be called `zone`. This
+proposal uses `clusterDomain` because it matches Kubelet's configuration field,
+and because `zone` is somewhat ambiguous (are we talking about the cluster zone?
+the namespace's zone?).
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
```

Comment:
If you want this to merge in 1.34, I would suggest filling these sections out also.

Reply:
This still holds.